packages/block-library/src/post-date/edit.js

```diff
@@ -155,7 +138,114 @@ export default function PostDateEdit( {
 					/>
 				</PanelBody>
 			</InspectorControls>
+
 			<div { ...blockProps }>{ postDate }</div>
 		</>
 	);
 }
+
+function is12HourFormat( format ) {
+	// To know if the time format is a 12 hour time, look for any of the 12 hour
+	// format characters: 'a', 'A', 'g', and 'h'. The character must be
+	// unescaped, i.e. not preceded by a '\'. Coincidentally, 'aAgh' is how I
+	// feel when working with regular expressions.
+	// https://www.php.net/manual/en/datetime.format.php
+	return /(?:^|[^\\])[aAgh]/.test( format );
+}
+
+function FormatSettings( { date, format, siteFormat, setFormat } ) {
```

Comment:
> I need to update the Comment Date block with the same changes. (Maybe the dropdown should be a block editor component? (Or a block supports? 😬))

Would it be worth moving this to the `block-editor` package so that the component can be re-used with the Comment Date block? I think that would make good sense for re-usability before exploring a date block support (which isn't a bad idea, but we might want to see if there's a good case for additional blocks before going too far further down that path?)

Reply:
Split it out into `DateFormatControl` in 102dd3867ef3e8327fd9e91616116e32b03f27ee 👍